features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -240,14 +241,72 @@ internal open class RumViewScope(
 
     @WorkerThread
     private fun onAddViewLoadingTime(event: RumRawEvent.AddViewLoadingTime, writer: DataWriter<Any>) {
-        if (stopped) return
-        val canAddViewLoadingTime = event.overwrite || viewLoadingTime == null
-        if (canAddViewLoadingTime) {
-            viewLoadingTime = event.eventTime.nanoTime - startedNanos
-            sendViewUpdate(event, writer)
-        } else {
-            // TODO RUM-6031 Add logs and telemetry here
+        val internalLogger = sdkCore.internalLogger
+        val canUpdateViewLoadingTime = !stopped && (viewLoadingTime == null || event.overwrite)
+        if (stopped) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.USER,
+                { NO_ACTIVE_VIEW_FOR_LOADING_TIME_WARNING_MESSAGE }
+            )
+            internalLogger.logApiUsage(
+                InternalTelemetryEvent.ApiUsage.AddViewLoadingTime(
+                    overwrite = event.overwrite,
+                    noView = false,
+                    noActiveView = true
+                )
+            )
+        }
+
+        if (canUpdateViewLoadingTime) {
+            updateViewLoadingTime(event, internalLogger, writer)
+        }
+    }
+
+    private fun updateViewLoadingTime(
+        event: RumRawEvent.AddViewLoadingTime,
+        internalLogger: InternalLogger,
+        writer: DataWriter<Any>
+    ) {
+        val viewName = key.name
+        val previousViewLoadingTime = viewLoadingTime
+        val newLoadingTime = event.eventTime.nanoTime - startedNanos
+        if (previousViewLoadingTime == null) {
+            internalLogger.log(
+                InternalLogger.Level.DEBUG,
+                InternalLogger.Target.USER,
+                { ADDING_VIEW_LOADING_TIME_DEBUG_MESSAGE_FORMAT.format(Locale.US, viewLoadingTime, viewName) }
+            )
+            internalLogger.logApiUsage(
+                InternalTelemetryEvent.ApiUsage.AddViewLoadingTime(
+                    overwrite = false,
+                    noView = false,
+                    noActiveView = false
+                )
+            )
+        } else if (event.overwrite) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.USER,
+                {
+                    OVERWRITING_VIEW_LOADING_TIME_WARNING_MESSAGE_FORMAT.format(
+                        Locale.US,
+                        viewName,
+                        previousViewLoadingTime,
+                        newLoadingTime
+                    )
+                }
+            )
+            internalLogger.logApiUsage(
+                InternalTelemetryEvent.ApiUsage.AddViewLoadingTime(
+                    overwrite = true,
+                    noView = false,
+                    noActiveView = false
+                )
+            )
         }
+        viewLoadingTime = newLoadingTime
```

Comment:
/blocking
 
here we are missing the happy case messaging and the telemetry.

`
View loading time added to the view
`

Reply:
the happy case is above (one of the if/else if) is the happy case.